SQL: Add server-wide default time zone config.

gianm opened the pull request:
It's useful when your users largely prefer a certain timezone.

Adds a new configuration parameter: `druid.sql.planner.sqlTimeZone`.

asdf2014 commented on sql/src/test/java/io/druid/sql/calcite/CalciteQueryTest.java:
```diff
@@ -169,6 +169,14 @@ public int getMaxQueryCount()
+    @Override
+    public DateTimeZone getSqlTimeZone()
+    {
+      return DateTimeZone.forID("America/Los_Angeles");
```
Maybe use `DateTimes.inferTzfromString` will be better here.

gianm replied:
I'll change this

jihoonson commented:
Even though I already approved this PR, please check the CI failure.

```
[ERROR] Forbidden method invocation: org.joda.time.DateTimeZone#forID(java.lang.String) [Uses default time zone, use factory methods from DateTimes and Intervals classes instead]
[ERROR]   in io.druid.sql.calcite.CalciteQueryTest$6 (CalciteQueryTest.java:177)
[ERROR] Scanned 44 (and 495 related) class file(s) for forbidden API invocations (in 0.13s), 1 error(s).
```

gianm commented:
@asdf2014 @jihoonson @nishantmonu51 Thanks for your comments -- I fixed the call to DateTimeZone.forID and added the new config parameter to the PR description. I also merged from master to try to help the tests to pass.
